### PR TITLE
Requests queue

### PR DESCRIPTION
**Pull request by llbartekll**
# Description

Resolves # (issue)

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

File: Example/WalletApp/PresentationLayer/Wallet/Main/MainPresenter.swift [hunk 1]
```diff
@@ -48,9 +49,14 @@ extension MainPresenter {
         
         interactor.sessionRequestPublisher
             .receive(on: DispatchQueue.main)
-            .sink { [unowned self] request, context in
-                router.dismiss()
-                router.present(sessionRequest: request, importAccount: importAccount, sessionContext: context)
+            .sink { [unowned self] (request, context) in
+                if let vc = UIApplication.currentWindow.rootViewController?.topController,
+                   vc.restorationIdentifier == SessionRequestModule.restorationIdentifier {
```
Comment (flypaper0) [inline Example/WalletApp/PresentationLayer/Wallet/Main/MainPresenter.swift:54]: kind of strange that type cast do not works. Also we can consider isKindOfClass

File: Example/WalletApp/PresentationLayer/Wallet/Main/MainPresenter.swift [hunk 2]
```diff
@@ -48,9 +49,14 @@ extension MainPresenter {
         
         interactor.sessionRequestPublisher
             .receive(on: DispatchQueue.main)
-            .sink { [unowned self] request, context in
-                router.dismiss()
-                router.present(sessionRequest: request, importAccount: importAccount, sessionContext: context)
+            .sink { [unowned self] (request, context) in
+                if let vc = UIApplication.currentWindow.rootViewController?.topController,
```
Comment (flypaper0) [inline Example/WalletApp/PresentationLayer/Wallet/Main/MainPresenter.swift:53]: guard?

File: Example/WalletApp/PresentationLayer/Wallet/Settings/SettingsPresenter.swift [hunk 3]
```diff
@@ -46,6 +47,7 @@ final class SettingsPresenter: ObservableObject {
         guard let account = accountStorage.importAccount?.account else { return }
         try await interactor.notifyUnregister(account: account)
         accountStorage.importAccount = nil
+        try await Web3Wallet.instance.cleanup()
```
Comment (flypaper0) [inline Example/WalletApp/PresentationLayer/Wallet/Settings/SettingsPresenter.swift:50]: 👍

File: Example/WalletApp/PresentationLayer/Wallet/Main/MainPresenter.swift (hunk 2 above)
Comment (llbartekll) [reply Example/WalletApp/PresentationLayer/Wallet/Main/MainPresenter.swift:53]: I knew you will point it out 😅

File: Example/WalletApp/PresentationLayer/Wallet/Main/MainPresenter.swift (hunk 1 above)
Comment (llbartekll) [reply Example/WalletApp/PresentationLayer/Wallet/Main/MainPresenter.swift]: u dont like restorationIdentifier approach?

**Review by flypaper0 — APPROVED** [review]

**Pull request merged**